### DISCUSSION TITLE
foxglove proto deserialize

### DISCUSSION THREAD

**Discussion by Muxindawang** [foxglove/mcap#1250] [General] [answered]
I have a ros1bag, and the topic content inside is obtained through protobuf serialization, with a type of std _ msgs/String. Can I directly write a script in Foxglove's user script to deserialize the data? If so, are there any examples that can be referenced
In the script, I can manipulate the tokenarray type data in ros1, but I don't know how to deserialize proto

![图片](https://github.com/user-attachments/assets/2ff45a01-be86-4f50-bab6-778dbee6ce48)
![图片](https://github.com/user-attachments/assets/073f0b9c-f896-459a-8747-76fcbdb230c1)

**Comment by jtbandes** [comment, marked as answer]
> I have a ros1bag, and the topic content inside is obtained through protobuf serialization, with a type of std _ msgs/String

Nope, this is not supported in Foxglove. Strings that are invalid UTF8 will probably not work. If they do, then you are free to interpret the string data in the user script however you like. But generally storing protobuf inside a ROS 1 bag is not recommended – MCAP is a better choice for that.
